tests/flagpole/test_conditions.py

```diff
@@ -113,22 +74,17 @@ def test_invalid_property_value(self):
         values = ["bar", "baz"]
         condition = InCondition(property="foo", value=values)
 
-        with pytest.raises(ConditionTypeMismatchException):
-            condition.match(context=EvaluationContext({"foo": []}), segment_name="test")
+        bad_context = ([1], {"k": "v"})
+        for attr_val in bad_context:
+            with pytest.raises(ConditionTypeMismatchException):
+                condition.match(context=EvaluationContext({"foo": attr_val}), segment_name="test")
 
         not_condition = NotInCondition(property="foo", value=values)
-        with pytest.raises(ConditionTypeMismatchException):
-            not_condition.match(context=EvaluationContext({"foo": []}), segment_name="test")
-
-    def test_valid_json_and_reparse(self):
-        values = [["foo", "bar"], [1, 2], [1.1, 2.2], []]
-        assert_valid_types(condition=InCondition, expected_types=values)
-        assert_valid_types(condition=NotInCondition, expected_types=values)
-
-    def test_invalid_value_type_parsing(self):
-        values = ["abc", 1, 2.2, True, None, ["a", 1], [True], [[]], [1, 2.2], [1.1, "2.2"]]
-        assert_invalid_types(condition=InCondition, invalid_types=values)
-        assert_invalid_types(condition=NotInCondition, invalid_types=values)
+        for attr_val in bad_context:
+            with pytest.raises(ConditionTypeMismatchException):
+                not_condition.match(
+                    context=EvaluationContext({"foo": attr_val}), segment_name="test"
+                )
```

Comment:
Main reason I went with the assertion util function approach before was because it made it explicit which invalid input failed when modifying condition logic. This is mostly fine, although it does make debugging just a _bit_ harder.